docs/API/API-functions.md

```diff
@@ -358,27 +358,74 @@ int srt_bind(SRTSOCKET u, const struct sockaddr* name, int namelen);
 
 Binds a socket to a local address and port. Binding specifies the local network
 interface and the UDP port number to be used for the socket. When the local 
-address is a form of `INADDR_ANY`, then it's bound to all interfaces. When the 
-port number is 0, then the port number will be system-allocated if necessary.
+address is a wildcard (`INADDR_ANY` for IPv4 or `in6addr_any` for IPv6), then
+it's bound to all interfaces (although see `SRTO_IPV6ONLY` and an additional
+information below for details about the wildcard address in IPv6).
+
+Binding is necessary for every socket to be used for communication. It can be
+bound implicitly, when the socket is being connected to a listening socket in
+the network using [`srt_connect`](#srt_connect) or a similar function that does
+the same, in which case the socket will be bound to a wildcard address of
+the IP version as specified in the `sockaddr::sa_family` field and port number
+0.  In other cases it must be bound explicitly using this function, or some
+other function that uses this functionality.
+
+When the port number is 0 (including the implicit binding case), then the port
+number will be system-allocated and can be determined after an explicit or
+implicit binding using [`srt_getsockname`](#srt_getsockname).
 
 This call is obligatory for a listening socket before calling [`srt_listen`](#srt_listen)
 and for rendezvous mode before calling [`srt_connect`](#srt_connect); otherwise it's 
 optional. For a listening socket it defines the network interface and the port where 
-the listener should expect a call request. In the case of rendezvous mode (when the
-socket has set [`SRTO_RENDEZVOUS`](API-socket-options.md#SRTO_RENDEZVOUS) to 
-true both parties connect to one another) it defines the network interface and port 
-from which packets will be sent to the peer, and the port to which the peer is 
-expected to send packets.
-
-For a connecting socket this call can set up the outgoing port to be used in the 
-communication. It is allowed that multiple SRT sockets share one local outgoing 
-port, as long as [`SRTO_REUSEADDR`](API-socket-options.md#SRTO_REUSEADDRS) 
-is set to *true* (default). Without this call the port will be automatically 
-selected by the system.
+the listener should expect a call request.
+
+In the case of rendezvous mode there's a pair of two endpoints (address-and-port
+specifications), let's name them A and B, and for both parties one endpoint is
+local and the other is remote one. The party, for which the A's address is
+local, should bind to the A address, and then connect to the B endpoint, and the
+other party the other way around. Both sockets must be set
+[`SRTO_RENDEZVOUS`](API-socket-options.md#SRTO_RENDEZVOUS) to *true* to make
+this connection possible.
```

Comment:
<img width="753" alt="image" src="https://user-images.githubusercontent.com/28537807/214648294-f2915107-e168-4b0f-b13e-477f89f6decb.png">

@ethouris Does this drawing reflect what you are trying to say in the text above?

Reply:
Rather something like this:
![image](https://user-images.githubusercontent.com/9600392/214790127-aff35bab-7bca-4613-9be7-b41f2fcf562c.png)

Reply:
Maybe this could be better.

```suggestion
In the case of rendezvous mode there are two parties that connect to one another. 
For every party there must be chosen a local binding endpoint (local address and port)
to which they expect connection from the opposite party. Let's say, we have a Party 1
that selects an endpoint A and a Party 2 that selects an endpoint B. In this case the Party 1
binds the socket to the endpoint A and then connects to the endpoint B, and the Party 2
the other way around. Both sockets must be set
[`SRTO_RENDEZVOUS`](API-socket-options.md#SRTO_RENDEZVOUS) to *true* to make
this connection possible.
```

Reply:
```suggestion
In rendezvous mode there are two parties that connect to one another. 
For each party you must designate a local binding endpoint (local address and port)
on which they expect the connection from the opposite party. Let's say Party 1 has 
endpoint A and Party 2 has endpoint B. In this case the Party 1 binds the socket to 
endpoint A and then connects to endpoint B. Party 2 does the reverse (binds the socket 
to endpoint B and then connects to endpoint A). Both sockets must be set  to *true* to make 
this connection possible (see [`SRTO_RENDEZVOUS`](API-socket-options.md#SRTO_RENDEZVOUS)).
```